addons/payment_mercado_pago/models/payment_transaction.py

```diff
@@ -0,0 +1,272 @@
+from odoo.addons.payment_mercado_pago.const import TRANSACTION_STATUS_MAPPING
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentTransaction(models.Model):
+    _inherit = 'payment.transaction'
+
+    def _get_specific_rendering_values(self, rendering_values):
+        """ Override of payment to return Mercado Pago specific rendering values.
+
+        Note: self.ensure_one() from `_get_rendering_values`.
+
+        :param dict rendering_values: The generic rendering values of the transaction.
+        :return: The dict of acquirer-specific processing values.
+        :rtype: dict
+        """
+        self.ensure_one()
+
+        res = super()._get_specific_rendering_values(rendering_values)
+        if self.provider != 'mercado_pago':
+            return res
+
+        # Create a preference to get a redirection url of Mercado Pago to initiate
+        # the payment flow.
+        preference = self._mercado_pago_create_preference()
+
+        if self.acquirer_id.state == 'test':
+            redirection_url = preference.get('sandbox_init_point')
+        else:
+            redirection_url = preference.get('init_point')
+
+        if not redirection_url:
+            raise ValidationError(
+                "Mercado Pago: " + _("no redirection url returned, cannot redirect.")
+            )
+
+        return {
+            'api_url': redirection_url,
+        }
+
+    def _mercado_pago_create_preference(self):
+        """ Create a preference for mercado pago to get the redirection url to begin the
+        payment flow.
+
+        Preferences are items or payments that can be created from MP, they answer with a
+        redirection url to pay such item.
+
+        :return: a dict with the answer of MP, in particular the redirection url
+        :rtype: dict
+        """
+        self.ensure_one()
+
+        # The back_url is a dict with the redirection urls for MP to return to.
+        back_url = None
+        if self.operation == 'online_redirect':
+            back_url = urls.url_join(self.get_base_url(), MercadoPagoController._return_url)
+
+        back_urls = {
+            'success': back_url,
+            'pending': back_url,
+            'failure': back_url,
+        } if back_url else {}
+
+        # URL to which MP can send notifications, MP does not provide enough information in their
+        # webhook notifications, thus we encode it in the return url.
+        notification_url = urls.url_join(
+            self.get_base_url(), f'{MercadoPagoController._webhook_url}/{self.id}'
+        )
+
+        items = [{
+            'title': self._mercado_pago_generate_payment_title(),
+            'quantity': 1,
+            'unit_price': self.amount,
+            'currency_id': self.currency_id.name,
+        }]
+
+        payer = {
+            'name': self.partner_name or {},
+            'email': self.partner_email or {},
+            'phone': {
+                'area_code': '',
```

Comment:
Why don't you provide the code ?
Are you sure it doesn't complain when the phone is partly specified ?